BackendWebApp/src/test/java/uk/ac/ebi/biostd/webapp/application/rest/service/PendingSubmissionServiceTest.java

```diff
@@ -0,0 +1,206 @@
+        final User user = newUser();
+
+        when(userDataService.findAllByUserAndTopic(user.getId(), TOPIC))
+                .thenReturn(ImmutableList.of(
+                        newUserData("1"),
+                        newUserData("2")
+                ));
+
+        PendingSubmissionListDto result = pendingSubmissionService.getSubmissionList(new PendingSubmissionListFiltersDto(), user);
+
+        assertThat(result.getSubmissions().size()).isEqualTo(2);
+        assertThat(result.getSubmissions().get(0).getAccno()).isEqualTo("1");
+        assertThat(result.getSubmissions().get(1).getAccno()).isEqualTo("2");
+    }
+
+    @Test
+    public void testDeleteSubmission() {
+        final User user = newUser();
+        final String accno = "1234";
+
+        doNothing().when(userDataService).deleteModifiedSubmission(user.getId(), accno);
+        pendingSubmissionService.deleteSubmissionByAccNo(accno, user);
+
+        verify(userDataService, times(1)).deleteModifiedSubmission(user.getId(), accno);
+    }
+
+    @Test
+    public void testUpdateSubmission() throws JsonProcessingException {
+        final User user = newUser();
+        final PendingSubmissionDto dto = newPendingSubmissionDto("1234");
+        final UserData userData = newUserData(dto);
+
+        when(userDataService.findByUserAndKey(user.getId(), dto.getAccno()))
+                .thenReturn(Optional.of(userData));
+
+        when(userDataService.update(eq(user.getId()), eq(dto.getAccno()), any(String.class), eq(TOPIC)))
+                .thenAnswer((Answer<UserData>) invocation -> {
+                    UserData ud = new UserData();
+                    ud.setData(invocation.getArgument(2));
+                    return ud;
+                });
+
+        Optional<PendingSubmissionDto> result = pendingSubmissionService.updateSubmission(dto.getAccno(), dto.getData(), user);
+
+        assertThat(result.isPresent()).isTrue();
+        result.ifPresent(
+                r -> {
+                    assertThat(r.getChanged()).isGreaterThan(dto.getChanged());
+                    assertThat(r.getData().toString()).isEqualTo(dto.getData().toString());
+                    assertThat(r.getAccno()).isEqualTo(dto.getAccno());
+                }
+        );
+    }
+
+    @Test
+    public void testCreateNewSubmission() {
+        final User user = newUser();
+        final JsonNode json = objectMapper.createObjectNode();
+
+        when(userDataService.update(eq(user.getId()), any(String.class), any(String.class), eq(TOPIC)))
+                .thenAnswer((Answer<UserData>) invocation -> {
+                    UserData ud = new UserData();
+                    ud.setData(invocation.getArgument(2));
+                    return ud;
+                });
+
+        Optional<PendingSubmissionDto> result = pendingSubmissionService.createSubmission(json, user);
+        assertThat(result.isPresent()).isTrue();
+        result.ifPresent(
+                r -> {
```

Comment:
I do not link that much this veroboty of using lamda when we already assert result optional is present, I would suggest to declared optional variable as `optioanlResult` and add new variable `result`

Reply:
:-) Okay. Done.